presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,506 @@
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_Area", POLYGON);
+        return geometry.getEsriGeometry().calculateArea2D();
+    }
+
+    @Description("Return a Geometry from Well-Known Text representation (WKT)")
+    @ScalarFunction("ST_GeometryFromText")
+    @SqlType(NAME)
+    public static Slice stGeometryFromText(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        return serialize(OGCGeometry.fromText(input.toStringUtf8()));
+    }
+
+    @SqlNullable
+    @Description("Return the Well-Known Text (WKT) representation of the geometry")
+    @ScalarFunction("ST_AsText")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice stAsText(@SqlType(NAME) Slice input)
+    {
+        return Slices.utf8Slice(deserialize(input).asText());
+    }
+
+    @Description("Returns the geometric center of a geometry")
+    @ScalarFunction("ST_Centroid")
+    @SqlType(NAME)
+    public static Slice stCentroid(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_Centroid", POLYGON);
+        SpatialReference reference = geometry.getEsriSpatialReference();
+        Envelope envelope = getEnvelope(geometry);
+        Point centroid = new Point((envelope.getXMin() + envelope.getXMax()) / 2, (envelope.getYMin() + envelope.getYMax()) / 2);
+        return serialize(OGCGeometry.createFromEsriGeometry(centroid, reference));
+    }
+
+    @Description("Return the coordinate dimension of the Geometry")
+    @ScalarFunction("ST_CoordDim")
+    @SqlType(StandardTypes.BIGINT)
+    public static long stCoordinateDimension(@SqlType(NAME) Slice input)
+    {
+        return deserialize(input).coordinateDimension();
+    }
+
+    @Description("Returns the inherent dimension of this Geometry object, which must be less than or equal to the coordinate dimension")
+    @ScalarFunction("ST_Dimension")
+    @SqlType(StandardTypes.BIGINT)
+    public static long stDimension(@SqlType(NAME) Slice input)
+    {
+        return deserialize(input).dimension();
+    }
+
+    @SqlNullable
+    @Description("Returns TRUE if the LINESTRING's start and end points are coincident")
+    @ScalarFunction("ST_IsClosed")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stIsClosed(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_IsClosed", LINE_STRING, MULTI_LINE_STRING);
+        MultiPath lines = (MultiPath) geometry.getEsriGeometry();
+        int pathCount = lines.getPathCount();
+        boolean result = true;
+        for (int i = 0; result && i < pathCount; i++) {
+            Point start = lines.getPoint(lines.getPathStart(i));
+            Point end = lines.getPoint(lines.getPathEnd(i) - 1);
+            result = result && end.equals(start);
+        }
+        return result;
+    }
+
+    @SqlNullable
+    @Description("Returns true if this Geometry is an empty geometrycollection, polygon, point etc")
```

Comment:
nit: perhaps, borrow the documentation from the standard

"Returns TRUE if a geometry value corresponds to the empty set"

It would be helpful to decide how to refer to TRUE value in the docs. Some comments use all-upper-case TRUE while others use lower-case true.